### PR TITLE
Support mom6 examples

### PR DESCRIPTION
**Pull request by mnlevy1981**
This addresses #121 by adding `manage_externals` to the repo and updating `Externals.cfg` to include `MOM6-examples` and `FMS`. I also added `mkmf`, necessary templates for building on cheyenne, and a `build_examples.sh` script to actually do the build.

@alperaltuntas and @gustavo-marques this is the PR I want to review at our meeting tomorrow afternoon

### CODE REVIEW

**Comment by alperaltuntas** [comment]
Thanks, @mnlevy1981 

Two quick comments before the meeting today:

 - Instead of checking out another copy of FMS, we should use the FMS copy under CESM/library/fms. The mkmf library should be able to use CESM/library/fms copy just fine.
 - Instead of including the entire MOM6-examples, I think we should include only a subset of the examples that we actively maintain and test.

**Comment by mnlevy1981** [comment]
Thanks for the feedback, @alperaltuntas ! These are both great suggestions, but we'll need to talk through implementation details this afternoon.

> * Instead of checking out another copy of FMS, we should use the FMS copy under CESM/library/fms. The mkmf library should be able to use CESM/library/fms copy just fine.

I included FMS because I was thinking some users might want to run `MOM6-examples` without checking out all of CESM. The current state of this PR allows them to check out `MOM_interface` by itself, run `./manage_externals/checkout_externals -o`, and have everything they need to build and run.

One idea I had was to create `Externals_MOM6-examples.cfg` with `MOM6-examples` as a required component and `FMS` as an optional component. Then CESM users would run `./manage_externals/checkout_externals -e Externals_MOM6-examples.cfg` while standalone `MOM_interface` users would need `./manage_externals/checkout_externals -oe Externals_MOM6-examples.cfg`.  I could modify `build_examples.sh` look to see if FMS is available in `$CESMROOT/library/fms` and fallback to `MOM6-examples/src/FMS` if it can't find the former. The downside to this proposal is that stand-alone `MOM_interface` users still need to checkout `MOM6` as well, which requires a second `./manage_externals/checkout_externals` call but maybe there's a way to link `Externals_MOM6-examples.cfg` back to the existing `Externals.cfg`?

Long story short, I'm happy to rearrange this but I think the goal should be a single call to `mange_externals` whether you are in a CESM sandbox or not and if we don't want CESM users to clone `FMS` then we need to think about how to organize the externals.

> * Instead of including the entire MOM6-examples, I think we should include only a subset of the examples, which we actively maintain and test.

This sounds good to me, but I'm not sure the best way to do it. Should we revive [NCAR/MOM6-examples](https://github.com/NCAR/MOM6-examples) and use it to curate a subset of examples? Or is [NCAR/MOM6-cases](https://github.com/NCAR/MOM6-cases/) a better choice? I think the answer is to use whatever repo @gustavo-marques is using for his testing.

**Comment by alperaltuntas** [comment]
Thanks! Though, these make me rethink the idea of supporting standalone MOM6 within MOM_interface. As its name gives away, MOM_interface is an interface between CESM and MOM6. In absence of CESM, we don't need MOM_interface at all.

So, I wonder, what may be the advantage of supporting standalone MOM6 within MOM_interface, as opposed to reviving and maintaining the MOM6-cases repository (an imitation of GFDL's MOM6-examples) that we specifically put together for our standalone MOM6 configurations: https://github.com/NCAR/MOM6-cases

**Comment by mnlevy1981** [comment]
> So, I wonder, what may be the advantage of supporting standalone MOM6 within MOM_interface, as opposed to reviving and maintaining the MOM6-cases repository

I think it's a huge boon for developers, because you only need a single checkout of your development path to test in both CESM and using MOM's `solo_driver`. And if more developers are running `solo_driver` tests, it (a) helps us catch issues with the build system (templates, mkmf, etc) much faster, and (b) makes it easier for users who want run the solo driver on the NCAR machines to find help getting set up. This PR was born out of my frustration at trying to develop the MARBL driver in the context of CESM while also trying to test the MARBL driver in a single column test case outside of CESM... a workflow that lets me do both from the same code base would be a huge help.

Also, if we are going to use the FMS library distributed with CESM when it is available, I think a good follow-up development would be to figure out how to use CIME's `configure` script to create the `Makefile` instead of `mkmf`. Then `MOM_interface` is an interface between CESM and MOM6 as well as an interface between all machines that support CESM and the solo driver and doing things like running the NCAR version of MOM6-examples on izumi should just work out of the box.

**Comment by alperaltuntas** [comment]
> I think it's a huge boon for developers, because you only need a single checkout of your development path to test in both CESM and using MOM's solo_driver.

If that's the main benefit, I feel like we shouldn't worry about the following side benefit:

> ... I was thinking some users might want to run MOM6-examples without checking out all of CESM.

I think we should require a full CESM checkout, so as to prevent unnecessary complexities and confusions.

> I could modify build_examples.sh look to see if FMS is available in $CESMROOT/library/fms and fallback to MOM6-examples/src/FMS if it can't find the former. 

This would prevent duplicate FMS checkouts only if the user first runs checkout_externals for the full CESM, but doesn't prevent it if the user first checks out standalone MOM6, and then the full CESM.

**Comment by mnlevy1981** [comment]
After discussion with @alperaltuntas and @gustavo-marques, this PR has been restructured. Rather than bring in `MOM6-examples` as an external, I copied most of the subdirectories in `MOM6-examples/ocean-only/` to `standalone/examples/`. The `mkmf` tools and the templates for building the model are in `standalone/mkmf/` and `standalone/templates/`, respectively. The script `standalone/build/build_ocean_only_examples.sh` will build the model.

This means we will need to stay in sync with GFDL's MOM6-examples by hand, but it avoids exposing the `ice_ocean` and `coupled` examples to users (since components necessary to run those examples are not available to CESM users).

So the workflow for running MOM's `solo_driver/` on cheyenne is:

1. clone a version of CESM with these changes to `MOM_interface`
2. use manage externals to check out `mom` and `fms`
   ```
   $ ./manage_externals/checkout_externals fms mom
   ```
3. `cd` to the `build/` directory and run the script
   ```
   $ cd components/mom/standalone/build/
   $ qcmd ./build_ocean_only_examples.sh
   ```

Then run the model by `cd`-ing to the proper directory, requesting an interactive node, and using `mpiexec_mpt`

```
$ cd ../examples/single_column/KPP
$ qinteractive
$ mpiexec_mpt dplace -s 1 ../../../build/intel/ocean_only/repro/MOM6
```

**Comment by mnlevy1981** [comment]
I'm happy to drop `ocean_only` from file and directory names, since we don't support any of the other executables.

**Comment by mnlevy1981** [comment]
@alperaltuntas I think fffb1ba has everything we talked about Tuesday afternoon - when you get a chance, can you look through it and let me know if anything else needs to be updated?

**Review by alperaltuntas — APPROVED** [review]
lgtm

**Review by gustavo-marques — APPROVED** [review]

**Comment by mnlevy1981** [comment]
I think all that's left is to add a script that runs each test case -- I'll put that together and push another commit today or tomorrow.

**Comment by mnlevy1981** [comment]
Okay, f689fab has the `run_all.sh` script -- it should be submitted to the queue, because a few of the jobs are not meant to run on a single task.

**Review by alperaltuntas — APPROVED** [review]
